core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -103,10 +118,12 @@ public CallOptions withCompression(@Nullable String compressorName) {
    * @param deadlineNanoTime the deadline in the clock as per {@link System#nanoTime()}.
    *                         {@code null} for unsetting the deadline.
    */
+  @Deprecated
   public CallOptions withDeadlineNanoTime(@Nullable Long deadlineNanoTime) {
-    CallOptions newOptions = new CallOptions(this);
-    newOptions.deadlineNanoTime = deadlineNanoTime;
-    return newOptions;
+    Deadline deadline = deadlineNanoTime != null
+        ? Deadline.after(deadlineNanoTime - System.nanoTime(), TimeUnit.NANOSECONDS)
```

Comment:
@buchgr, returning negative numbers came up in the initial commit of `timeRemaining()`, but @louiscryan chose not to return negatives. I talked with @louiscryan about what you are experiencing, and we're both in favor of having `timeRemaining()` return negatives. So let's go that way. It won't be 100% accurate, but the edge case is racy no matter how you look at it.